sdk/eventhubs/azure-messaging-eventhubs/src/producer_client.cpp

```diff
@@ -87,6 +88,14 @@ namespace Azure { namespace Messaging { namespace EventHubs {
       // Throw an exception about the error we just received.
       throw Azure::Messaging::EventHubs::_detail::EventHubsExceptionFactory::
           CreateEventHubsException(std::get<1>(result));
+#elif ENABLE_RUST_AMQP
+      if (result)
+      {
+        throw Azure::Messaging::EventHubs::_detail::EventHubsExceptionFactory::
+            CreateEventHubsException(result);
+      }
+      return true;
```

Comment:
Should this be the inverse? That is:
`if(result) return true;`

Reply:
The result is an error in the rust amqp case (it's a tuple of some internal uAMQP state and an error in the uamqp state). If the error is true, there is an error, and if there is an error, we're supposed to throw an exception.